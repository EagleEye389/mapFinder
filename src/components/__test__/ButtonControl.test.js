import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";

import Button from "../customButton/ButtonControl";

let component;
let buttonText = "test";
let buttonType = "primary-button";
let a = 5;
const test = () => a++;

describe("Button component is working properly", () => {
  beforeEach(() => {
    component = shallow(
      <Button
        label={buttonText}
        type={buttonType}
        handleClick={test}
        disableCheck={true}
      />
    );
  });

  it("should render button and show text properly", () => {
    expect(component.props().children).toBe(buttonText);
  });

  it("should render button and check class is present", () => {
    expect(component.hasClass(buttonType)).toBe(true);
  });

  it("should render button and check click is working", () => {
    component.find("button").simulate("click");
    expect(a).toBe(6);
  });
});

describe("Button snapshot", () => {
  it("checks enable button snapshot", () => {
    const tree = renderer
      .create(
        <Button
          label={buttonText}
          type={buttonType}
          handleClick={test}
          disableCheck={true}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("checks disable button snapshot", () => {
    const tree = renderer
      .create(
        <Button
          label={buttonText}
          type={buttonType}
          handleClick={test}
          disableCheck={false}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
