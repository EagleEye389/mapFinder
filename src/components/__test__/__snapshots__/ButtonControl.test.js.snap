// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button snapshot checks disable button snapshot 1`] = `
<button
  className="primary-button"
  disabled={false}
  onClick={[Function]}
>
  test
</button>
`;

exports[`Button snapshot checks enable button snapshot 1`] = `
<button
  className="primary-button"
  disabled={true}
  onClick={[Function]}
>
  test
</button>
`;
